### PR TITLE
log2

### PR DESCRIPTION
**Pull request by mohanadtalat91**
Issue #5964

### CODE REVIEW

**Comment by mohanadtalat91** [comment]
Hi @CatB1t, 👋😊
Could you please review my PR, feel free to ask for any updates. Thanks in advanced

File: ivy/array/elementwise.py [hunk 1]
```diff
@@ -1243,6 +1243,24 @@ def log2(self: ivy.Array, *, out: Optional[ivy.Array] = None) -> ivy.Array:
...
+        --------
+        Using :code:`ivy.Array` instance method:
+        >>> x = ivy.array([4.0, 1, -0.0, -5.0])
+        >>> y = x.log2()
+        >>> print(y)
+        ivy.array([2., 0., -inf, nan])
+        >>> x = ivy.array([float('nan'), -4.0, -0.0, 1.0, 4.0, float('+inf')])
+        >>> y = x.log2()
+        >>> print(y)
+        ivy.array([nan, nan, -inf, 0., 2., inf])
```
Comment (CatB1t) [inline ivy/array/elementwise.py:1257]: Space after output

File: ivy/array/elementwise.py [hunk 2]
```diff
@@ -1243,6 +1243,24 @@ def log2(self: ivy.Array, *, out: Optional[ivy.Array] = None) -> ivy.Array:
...
             in ``self``. The returned array must have a real-valued floating-point
             data type determined by :ref:`type-promotion`.
+
+	Examples
+        --------
+        Using :code:`ivy.Array` instance method:
+        >>> x = ivy.array([4.0, 1, -0.0, -5.0])
+        >>> y = x.log2()
+        >>> print(y)
+        ivy.array([2., 0., -inf, nan])
```
Comment (CatB1t) [inline ivy/array/elementwise.py:1253]: Space after output

File: ivy/functional/ivy/elementwise.py [hunk 3]
```diff
@@ -3568,6 +3568,36 @@ def log2(
...
+                       [+0, -1.0, -5, float('-inf')]])
+    >>> y = ivy.log2(x)
+    >>> print(y)
+    ivy.array([[nan, 0, 2.807, inf],
+               [-inf, nan, nan, nan]])
+    With :code:`ivy.NativeArray` input:
+    >>> x = ivy.native_array([4.56, float('-inf')])
+    >>> y = ivy.log2(x)
+    >>> print(y)
+    ivy.array([2.189, nan])
```
Comment (CatB1t) [inline ivy/functional/ivy/elementwise.py:3588]: New line after output

File: ivy/functional/ivy/elementwise.py [hunk 4]
```diff
@@ -3568,6 +3568,36 @@ def log2(
...
+    >>> x = ivy.array([4.0, 1, -0.0, -5.0])
+    >>> y = ivy.log2(x)
+    >>> print(y)
+    ivy.array([2.0, 0, -inf, nan])
+    >>> x = ivy.array([[float('nan'), 1, 7.0, float('+inf')],\
+                       [+0, -1.0, -5, float('-inf')]])
+    >>> y = ivy.log2(x)
+    >>> print(y)
+    ivy.array([[nan, 0, 2.807, inf],
+               [-inf, nan, nan, nan]])
```
Comment (CatB1t) [inline ivy/functional/ivy/elementwise.py:3583]: New line after output

**Review by CatB1t — CHANGES_REQUESTED** [review]
Just few changes, we need to add a new lien after the example output. could you please fix that?

**Comment by mohanadtalat91** [comment]
> Just few changes, we need to add a new lien after the example output. could you please fix that?

I've fixed these changes. @CatB1t

File: ivy/array/elementwise.py [hunk 5]
```diff
@@ -1243,6 +1243,29 @@ def log2(self: ivy.Array, *, out: Optional[ivy.Array] = None) -> ivy.Array:
             an array containing the evaluated base ``2`` logarithm for each element
             in ``self``. The returned array must have a real-valued floating-point
             data type determined by :ref:`type-promotion`.
+
+	Examples
+        --------
+
+        Using :code:`ivy.Array` instance method:
```
Comment (CatB1t) [inline ivy/array/elementwise.py:1250]: Could you provide a simple example by only using a 0-dim input? for example `ivy.array(10)`

**Review by CatB1t — CHANGES_REQUESTED** [review]
Just one small change, then we're good to merge!

**Comment by mohanadtalat91** [comment]
> 

Hello @CatB1t👋 , the 0-dim example has been added.

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @mohanadtalat91, everything looks great! your changes seems to be causing the `lint / check-formatting` test to fail though, could please take a look and get that fixed before merging? Thanks! 😄

**Comment by mohanadtalat91** [comment]
> Hello @mohanadtalat91, everything looks great! your changes seems to be causing the `lint / check-formatting` test to fail though, could please take a look and get that fixed before merging? Thanks! 😄

Hi @CatB1t, I've resolved failure of this test, all `black & flake8` are passed at my local, but still failling here at `lint / check-formatting (pull_request)` !!
  
And here's the screenshot 
ivy

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @mohanadtalat91, I've fixed the linting and styling errors for you, but the output of examples seems to be invalid, can you double check why is it fail and get those fixed? Thanks!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by mohanadtalat91** [comment]
> Hello @mohanadtalat91, I've fixed the linting and styling errors for you, but the output of examples seems to be invalid, can you double check why is it fail and get those fixed? Thanks!

Yeah, sure.
Thanks

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by mohanadtalat91** [comment]
> 

Continue

**Comment by CatB1t** [comment]
Hello @mohanadtalat91 , `test-docstrings` and `test-numpy-style-docstrings` are still failing, can you have a look at what's causing them to fail and fix those? Thanks! 😄

**Review by CatB1t — CHANGES_REQUESTED** [review]
Hello @mohanadtalat91 , `test-docstrings` and `test-numpy-style-docstrings` are still failing, can you have a look at what's causing them to fail and fix those? Thanks! 😄

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**